### PR TITLE
Feat/utility button cleanup

### PR DESCRIPTION
**Pull request by justindwyer6**
The utility buttons needed some text to be easier to understand, and some needed feedback on success.

### CODE REVIEW

File: src/hooks/useBooleanTimeout.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+
+const useBooleanTimeout = (timeout = 1000) => {
```
Comment (abrad45) [inline src/hooks/useBooleanTimeout.js:3]: did you write this or copy from somewhere? I like adding a link in for copied code so you can refer back to the source, see potential updates, etc.

File: src/components/App/App.js [hunk 2]
```diff
@@ -40,28 +40,30 @@ const App = () => {
       <Rules rulesOpen={areRulesOpen} />
       <Header />
       <div className="utilities">
+        <DiceRoller />
         <IconButton
-          name="Open rules"
+          name="Rules"
           imgSrc={rulesIconSrc}
           onClickFunction={() => toggleRules()}
```
Comment (abrad45) [inline src/components/App/App.js:47]: I know you didn't change anything here, but generally, when building components, using the same properties as the react defaults is preferred. A developer unfamiliar with `IconButton` would assume this prop would be named `onClick` but it's not. This reduces cognitive load

File: src/components/IconButton/index.jsx [hunk 3]
```diff
@@ -33,10 +50,12 @@ const IconButton = ({
...
     >
-      <img
-        src={confirmationRequested ? questionMarkIcon : imgSrc}
-        alt={name}
-      />
+      <img src={getIcon()} alt={name} />
+      <span>
+        {confirmationRequested
+          ? "Confirm"
+          : name.replace(" ", "\u00A0")}
```
Comment (abrad45) [inline src/components/IconButton/index.jsx:57]: why do we need to replace spaces in the names? This is the kind of line that's probably worth either refactoring out of the app or leaving a comment to explain.

File: src/components/IconButton/index.jsx [hunk 4]
```diff
@@ -1,28 +1,45 @@
...
+
+  const getIcon = () => {
+    if (confirmationRequested) {
+      return questionMarkIcon;
+    }
+    if (giveSuccessFeedback && actionSucceeded) {
+      return checkMarkIcon;
     }
+    return imgSrc;
   };
```
Comment (abrad45) [inline src/components/IconButton/index.jsx:43]: I don't know that you can do this in a less complex way other than by not overloading the icon in the first place. For instance, you could redo the entire IconButton to render either a button or a confirmation or a success, and then the logic there is handled in the other components.

Put another way: right now, you're keying off of a few values in a few places and rendering an icon based on one conditional, text based on another, etc. Why not do that once, and render a new component depending on the outcome?

You know, or this is fine honestly :) I'd say `handleClick` would benefit more from tweaking than `getIcon`

**Review by abrad45 — APPROVED** [review]

File: src/hooks/useBooleanTimeout.js (hunk 1 above)
Comment (justindwyer6) [reply src/hooks/useBooleanTimeout.js:3]: Nope, I wrote this myself!

File: src/components/IconButton/index.jsx (hunk 3 above)
Comment (justindwyer6) [reply src/components/IconButton/index.jsx:57]: Good call, I'll pull this out into its own function. It's replacing spaces with the non-breaking space character.

File: src/components/IconButton/index.jsx (hunk 4 above)
Comment (justindwyer6) [reply src/components/IconButton/index.jsx:43]: Ok cool, then I'll leave as-is and maybe revisit later if I feel like it. 😄

**Pull request merged**